kernel/src/callback.rs

```diff
@@ -13,12 +13,16 @@ pub struct AppId {
 /// These IDs are used to identify which kernel container is being accessed.
 const KERNEL_APPID_BOUNDARY: usize = 100;
 
-impl AppId {
-    pub fn new(idx: usize) -> AppId {
```

Comment:
why not `pub (crate) const fn new(idx: usize) -> AppId` ?

`pub (crate)` - it is only accessable to the current crate and its submodules!

Reply:
How do you find these things?!? This is definitely what we want, and leave it to rust to come up with yet another new syntax.

Reply:
Haha, I have been reading plenty of rust code lately and sometimes I pick up a couple of things.

BTW, there is some docs about it https://doc.rust-lang.org/reference/visibility-and-privacy.html but this is stabilized so I wouldn't expect it to change since Rust strives to be backward-compatible but we'll see!